compiler/noirc_evaluator/src/brillig/mod.rs

```diff
@@ -43,7 +45,11 @@ impl Brillig {
                 self.ssa_function_to_brillig.get(&function_id).map(Cow::Borrowed)
             }
             // Procedures are compiled as needed
-            LabelType::Procedure(procedure_id) => Some(Cow::Owned(compile_procedure(procedure_id))),
+            LabelType::Procedure(procedure_id) => {
+                let mut artifact = compile_procedure(procedure_id);
+                artifact.procedure = Some(procedure_id);
+                Some(Cow::Owned(artifact))
```

Comment:
Would it make sense to assign `procedure` inside `compile_procedure`? I see it's also called in a different place but I don't know if that other place should have it set or not.

Reply:
The other place is just a test you are right it is more natural in `compile_procedure`. I moved the setting of the procedure setting to the instantiation of the Brillig context and artifact in `new_for_procedure`.